### PR TITLE
Hide users page from anyone who isn't an admin & add object status modal for editor

### LINKED ISSUE #8: Add meatball menu item "Change Status"
As an Editor/Admin, I want to be able to release a Learning Object through an Interaction with the UI so that I can give the public access to reviewed materials.

### LINKED ISSUE #10: Modal for changing status

### LINKED ISSUE #13: Extend Authguard to give lead/curator access to admin dashboard
As a lead/curator, I want to be able to use the admin dashboard.

### LINKED ISSUE #16: Hide users page from anyone who isn't an Admin

### PR DESCRIPTION
**Pull request by jbarlow96**
Whenever an editor or curator go on the admin dashboard, they only see the objects page. Admin still sees both. An object status modal for editor that contains "unreleased" and "released" has been implemented too. The learning objects and users still don't load on dashboard for curators/editors due to learning object services.

closes #16 
closes #13 
closes #8 
closes #10

### CODE REVIEW

File: src/app/dashboard/sidebar/sidebar.component.ts [hunk 1]
```diff
@@ -17,15 +17,17 @@ export class SidebarComponent implements OnInit, OnDestroy {
   email: string;
   profileImg: string;
   subscriptions: Subscription[] = [];
+  isUsersPageViewable = false;
   constructor(
     private router: Router,
-    private authService: AuthService,
+    public authService: AuthService,
```
Comment (DonSeannelly) [inline src/app/dashboard/sidebar/sidebar.component.ts]: Why was this changed to public? I don't see it being used anywhere in the template.

File: src/app/dashboard/sidebar/sidebar.component.html [hunk 2]
```diff
@@ -6,8 +6,8 @@
     <div class="profile-image" [style.background-image]="'url(' + profileImg + ')'"></div>
     <div class="welcome">Welcome {{ name | titlecase }}</div>
   </div>
-  <div class="menu-item " [ngClass]="{active: activeLink==='' }" (click)="navigateTo('')">USERS</div>
-  <div class="menu-item " [ngClass]="{active: activeLink==='objects' }" (click)="navigateTo('objects')">OBJECTS</div>
+  <div class="menu-item " [ngClass]="{active: activeLink==='' }" (click)="navigateTo('')">OBJECTS</div>
+  <div class="menu-item " *ngIf="isUsersPageViewable" [ngClass]="{active: activeLink==='users' }" (click)="navigateTo('users')">USERS</div>
```
Comment (DonSeannelly) [inline src/app/dashboard/sidebar/sidebar.component.html]: It looks like the order of menu items was switched. Was this done on purpose?

Comment (jbarlow96) [reply src/app/dashboard/sidebar/sidebar.component.html]: Yes this was done on purpose. Since users would automatically load up once a user would launch the dashboard, I changed it to object since that is accessible to each of the roles that can access the admin dashboard.

File: src/app/dashboard/sidebar/sidebar.component.ts (hunk 1 above)
Comment (jbarlow96) [reply src/app/dashboard/sidebar/sidebar.component.ts]: You're right, I believe there was no reason to change this to public. I will change it back now!

File: src/app/core/auth.service.ts [hunk 3]
```diff
@@ -116,4 +126,41 @@ export class AuthService {
...
+   */
+  public hasEditorAccess(): boolean {
+    return this.group.getValue() === AUTH_GROUP.EDITOR;
+  }
+
+  /**
+   * Identifies if the current logged in user has curator priviledges.
+   */
+  public hasCuratorAccess(): boolean {
+    return this.group.getValue() === AUTH_GROUP.CURATOR;
```
Comment (nrwinner) [inline src/app/core/auth.service.ts]: Can we add a FIXME since this is a temporary measure fix?

File: src/app/core/auth-guard.service.ts [hunk 4]
```diff
@@ -11,7 +11,9 @@ import { User } from '@cyber4all/clark-entity';
 import { environment } from '@env/environment';
 
 const enum ACCESS_GROUPS {
```
Comment (nrwinner) [inline src/app/core/auth-guard.service.ts]: Instead of declaring this twice, could we import this from the auth service? This way, we have a single source of truth.

File: src/app/core/auth.service.ts (hunk 3 above)
Comment (DonSeannelly) [reply src/app/core/auth.service.ts]: Perhaps something like this?

```suggestion
    // FIXME: This should be expanded to identify if the user has curator access to a specific collection
    return this.group.getValue() === AUTH_GROUP.CURATOR;
```

File: src/app/core/auth.service.ts [hunk 5]
```diff
@@ -116,4 +126,41 @@ export class AuthService {
...
+   * The highest priority group will be assigned.
+   */
+  private assignUserToGroup() {
+    if (this.user['accessGroups']) {
+      if (this.user['accessGroups'] && this.user['accessGroups'].includes('admin')) {
+        this.group.next(AUTH_GROUP.ADMIN);
+      } else { (this.user['accessGroups'].includes('editor'));
+          this.group.next(AUTH_GROUP.EDITOR);
+      }
+    } else { (this.user['accessGroups'].includes('curator'));
```
Comment (DonSeannelly) [inline src/app/core/auth.service.ts]: I'm not following the syntax here. It looks like perhaps this was meant to be an else-if?

Comment (DonSeannelly) [reply src/app/core/auth.service.ts]: We would also need a FIXME here as this logic will change once the curator string is stored as `curator@COLLECTION_NAME`

File: src/app/core/auth-guard.service.ts [hunk 6]
```diff
@@ -38,7 +34,17 @@ export class AuthGuard implements CanActivate {
         user => {
           if (
             user.accessGroups &&
-            (<ACCESS_GROUPS[]>user.accessGroups).includes(ACCESS_GROUPS.ADMIN)
+            (<AUTH_GROUP[]>user.accessGroups).includes(AUTH_GROUP.ADMIN)
```
Comment (gshaw1997) [inline src/app/core/auth-guard.service.ts]: Condense this `if else if else` chain to just `if else` by using the `||` operator for cases that will return the same value. Increases readabillity.

File: src/app/core/auth.service.ts [hunk 7]
```diff
@@ -11,11 +11,20 @@ import { Http, Headers, ResponseContentType } from '@angular/http';
...
+  ADMIN = 'admin'
+}
+
+
 @Injectable()
 export class AuthService {
   user: User = undefined;
   httpHeaders = new HttpHeaders();
   isLoggedIn = new BehaviorSubject<boolean>(false);
+  group = new BehaviorSubject<AUTH_GROUP>(AUTH_GROUP.CURATOR);
```
Comment (gshaw1997) [inline src/app/core/auth.service.ts]: Why is this a `BehaviorSubject` initialized with `AUTH_GROUP.CURATOR`?

Comment (jbarlow96) [reply src/app/core/auth.service.ts]: I initialized it with AUTH_GROUP.CURATOR because the lowest value enum member out of curator, editor and admin. Is there a better way of doing this?

File: src/app/core/auth-guard.service.ts [hunk 8]
```diff
@@ -38,9 +34,11 @@ export class AuthGuard implements CanActivate {
         user => {
           if (
             user.accessGroups &&
-            (<ACCESS_GROUPS[]>user.accessGroups).includes(ACCESS_GROUPS.ADMIN)
+            (<AUTH_GROUP[]>user.accessGroups).includes(AUTH_GROUP.ADMIN) || user.accessGroups &&
+            (<AUTH_GROUP[]>user.accessGroups).includes(AUTH_GROUP.EDITOR) || user.accessGroups &&
+            (<AUTH_GROUP[]>user.accessGroups).includes(AUTH_GROUP.CURATOR)
           ) {
-            return true;
+              return true;
```
Comment (DonSeannelly) [inline src/app/core/auth-guard.service.ts]: Looks like some extra whitespace was added here.
